src/torchcodec/_core/ops.py

```diff
@@ -212,6 +212,22 @@ def get_frames_at_indices(
     return _get_frames_at_indices_tensor_input(decoder, frame_indices=frame_indices)
 
 
+def get_frames_by_pts(
+    decoder: torch.Tensor, *, timestamps: Union[torch.Tensor, list[float]]
+) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+    if isinstance(timestamps, torch.Tensor):
+        # Ensure indices is the correct dtype (float64)
+        timestamps = timestamps.to(torch.float64)
+    else:
+        # Convert list to tensor for dispatch
+        try:
+            timestamps = torch.tensor(timestamps, dtype=torch.float64)
+        except (ValueError, TypeError):
+            # Type validation in C++ layer
+            pass
```

Comment:
I think we should try to avoid silencing exceptions here, can you share more about the kind of errors we're expecting?

Reply:
Thank you for the feedback!